[TECH] Migrer vers nuxt 3

matthiasferraina opened the pull request:
## :unicorn: Problème
Nuxt 2 est devenu obsolète le 31 décembre 2023 (https://v2.nuxt.com/fr/lts/). Afin de rester à jour, il est nécessaire d'effectuer la montée de version vers nuxt 3

## :robot: Proposition
Faire la montée de version

## :rainbow: Remarques
L'utilisation de l'api composable n'a pas été faite à chaque fois. L'accent a été mis sur la conservation d'une application iso fonctionnelle. 

## :100: Pour tester

Jeyffrey committed:
second

Jeyffrey committed:
add public images

Jeyffrey committed:
fix PageBanner without modal gestion

Jeyffrey committed:
add translations

Jeyffrey committed:
functional news page

Jeyffrey committed:
improve locale usage

Jeyffrey committed:
display latest news

Jeyffrey committed:
display nav

Jeyffrey committed:
fetch news with locale

Jeyffrey committed:
add language switcher

Jeyffrey committed:
rectify some links

Jeyffrey committed:
fonctional modal

Jeyffrey committed:
fix page banner

Jeyffrey committed:
add dynamic page file

Jeyffrey committed:
article slice with vue setup

Jeyffrey committed:
add main footer

Jeyffrey committed:
add news item single

Jeyffrey committed:
init shared folder

Jeyffrey committed:
config imports from shared folder

Jeyffrey committed:
shared translations

Jeyffrey committed:
init pix-pro folder

Jeyffrey committed:
rename environment url hook

Jeyffrey committed:
add useEnvironmentUrl test

Jeyffrey committed:
fix layout nav

Jeyffrey committed:
update sites ports

Jeyffrey committed:
shared nuxt config

Jeyffrey committed:
fix homepage data fetching

Jeyffrey committed:
move translations files into nuxt projets to prevent prod problems

Jeyffrey committed:
add public images to pix-pro

Jeyffrey committed:
fix locale switcher

Jeyffrey committed:
rename shared locale-switcher component

Jeyffrey committed:
fix footer links

Jeyffrey committed:
rename shared footer-slice-zone component

Jeyffrey committed:
add 404 error page

Jeyffrey committed:
handle all pages types

Jeyffrey committed:
filter pages query by site

Jeyffrey committed:
improve shared footer and navigation queries

Jeyffrey committed:
add working languages switcher

Jeyffrey committed:
add working main navigation

Jeyffrey committed:
use the right nuxt-config in pix-site

Jeyffrey committed:
init playwright tests for header

Jeyffrey committed:
fix pages query depending of the current site env

Jeyffrey committed:
use nuxt-3 layers for shared folder

Jeyffrey committed:
fix PageBanner videoClass build

Jeyffrey committed:
get SITE env variable from nuxt app configuration

Jeyffrey committed:
move public folder only in shared directory

Jeyffrey committed:
add vite svg-loader plugin

Jeyffrey committed:
use svg in video button

Jeyffrey committed:
improve locale-switcher a11y and traductions

Jeyffrey committed:
fix language switcher e2e test

Jeyffrey committed:
improve useEnvironment composable for local homepage link

Jeyffrey committed:
add main navigation e2e test

Jeyffrey committed:
fix some errors for initial deployment

Jeyffrey committed:
test package.json shit install

Jeyffrey committed:
remove shared package.json

yannbertrand committed:
chore: pin node version

yannbertrand committed:
chore: add nginx conf

yannbertrand committed:
refactor: move buildpack and nginx conf

yannbertrand committed:
fix: nginx conf

Jeyffrey committed:
tech: update news slug file

Jeyffrey committed:
tech: cancel use of different domains in i18n config

Jeyffrey committed:
feat: add LocaleChoice component for pix-site

Jeyffrey committed:
chore: add deploiement files for pix-site

Jeyffrey committed:
update urls in prismic-rich-text

Jeyffrey committed:
tech: modify layout shift for locale-choice page

Jeyffrey committed:
feat: add a prismic-rich-text serializer for document hyperlinks

Jeyffrey committed:
tech: rename composables tests folder

Jeyffrey committed:
tech: init footer e2e test

Jeyffrey committed:
feat: prevent locale-choice page for urls with locale

Jeyffrey committed:
feat: add favicons

Jeyffrey committed:
tech: remove PixImage component

matthiasferraina committed:
add e2e test

Jeyffrey committed:
feat: use nuxt-link component for inner links

Jeyffrey committed:
tech: update locale-choice e2e tests

Jeyffrey committed:
tech: LocaleSwitcher now updates locale cookie

Jeyffrey committed:
tech: add homepage E2E test

Jeyffrey committed:
rename NavigationSliceZone component name

Jeyffrey committed:
feat: init main navigation v2

clarissedmn committed:
fix: locale switcher toggle

Jeyffrey committed:
feat: page title in document head

Jeyffrey committed:
feat: add main navigation e2e test

Jeyffrey committed:
feat: add skip-link component

Jeyffrey committed:
tech: add useMainNavigationLinks composable

clarissedmn committed:
feat: add desktop menu v2

clarissedmn committed:
fix responsive in mobile and allow tab on it

matthiasferraina committed:
prepare repo for migration to nuxt 3

pix-bot-github commented:
Une fois les applications déployées, elles seront accessibles via les liens suivants :
  * [Pix Site (.fr)](https://site-pr590.review.pix.fr)
  * [Pix Site (.org)](https://site-pr590.review.pix.org)
  * [Pix Pro (.fr)](https://pro-pr590.review.pix.fr)
  * [Pix Pro (.org)](https://pro-pr590.review.pix.org)

Les variables d'environnement seront accessibles sur scalingo https://dashboard.scalingo.com/apps/osc-fr1/pix-site-review-pr590/environment

clarissedmn committed:
feat: remove feature toggle isNewMenuAvailable

clarissedmn committed:
feat: improve accessibility

Jeyffrey committed:
fix: PartnersLogos if image exists

matthiasferraina committed:
setup build

matthiasferraina committed:
disable locale home for FR domain for pix-site

matthiasferraina committed:
setup build for pix-pro

matthiasferraina committed:
feat: generate review app env with appropriate domain url

Co-authored-by: Vincent Hardouin <vincent.hardouin@pix.fr>

matthiasferraina committed:
fix: locale switcher

Co-authored-by: Vincent Hardouin <vincent.hardouin@pix.fr>

VincentHardouin committed:
feat: keep only fr-fr on SITE_DOMAIN=FR

Co-authored-by: Matthias Ferraina <matthias.ferraina@pix.fr>

VincentHardouin committed:
feat: remove fr-fr on SITE_DOMAIN=ORG

Co-authored-by: Matthias Ferraina <matthias.ferraina@pix.fr>

matthiasferraina committed:
feat: handle all links

Co-authored-by: Vincent Hardouin <vincent.hardouin@pix.fr>

VincentHardouin committed:
feat: redirect with https

Co-authored-by: Matthias Ferraina <matthias.ferraina@pix.fr>

VincentHardouin committed:
feat: adapt ci to nuxt3 project dir

Co-authored-by: Matthias Ferraina <matthias.ferraina@pix.fr>

VincentHardouin committed:
feat: add index local choice page in pix-site

Co-authored-by: Matthias Ferraina <matthias.ferraina@pix.fr>

VincentHardouin committed:
feat: remove index page when SITE_DOMAIN=FR

VincentHardouin committed:
feat: improve ci workflow

matthiasferraina committed:
feat: display easiware forms

matthiasferraina committed:
feat: generate routes that are not linked in prismic documents

Co-authored-by: Vincent Hardouin <vincent.hardouin@pix.fr>

matthiasferraina committed:
feat: add locale suggestion banner to layout

Co-authored-by: Vincent Hardouin <vincent.hardouin@pix.fr>

VincentHardouin committed:
feat: disable detect browser language

Co-authored-by: Matthias Ferraina <matthias.ferraina@pix.fr>

VincentHardouin committed:
feat: run only org site with playwright

Co-authored-by: Matthias Ferraina <matthias.ferraina@pix.fr>

matthiasferraina committed:
feat: enable query params configuration for 123formbuilder forms

VincentHardouin committed:
chore: update @nuxtjs/i18n to 8.0.0

VincentHardouin committed:
feat: not used existing default locale to prevent rootRedirect

Co-authored-by: Matthias Ferraina <matthias.ferraina@pix.fr>

VincentHardouin committed:
feat: add reachable locale concept to distinct with i18n locales

- reachable locale = all pix locales in all sites
- locales = i18n locales present in current site

Co-authored-by: Matthias Ferraina <matthias.ferraina@pix.fr>

VincentHardouin committed:
refactor: remove useless scripts

VincentHardouin committed:
refactor: duplicate i18n config in each site

- handle nuxt prepare after npm ci when SITE_DOMAIN is not defined.

VincentHardouin committed:
feat: use localeCookie to redirect in index page

Co-authored-by: Matthias FERRAINA <matthias.ferraina@pix.fr>

matthiasferraina committed:
feat: add matomo

VincentHardouin committed:
fix e2e tests

matthiasferraina committed:
test: migrate test for link-resolver.js

matthiasferraina committed:
test: migrate test for document-fetcher.js

matthiasferraina committed:
test: migrate test for filter-nuxt-pages service

matthiasferraina committed:
sr: delete unused files

VincentHardouin committed:
feat: add fr-fr server for e2e

matthiasferraina committed:
test: fix end to end tests
Co-authored-by: Eric Lim <eric.lim@pix.fr>
Co-authored-by: Vincent Hardouin <vincent.hardouin@pix.fr>

matthiasferraina committed:
test: use another resource to avoid flaky

Co-authored-by: Eric Lim <eric.lim@pix.fr>
Co-authored-by: Vincent Hardouin <vincent.hardouin@pix.fr>

er-lim committed:
fix: correct use show out of france banner composable

Co-authored-by: Vincent Hardouin <vincent.hardouin@pix.fr>
Co-authored-by: Matthias Ferraina <matthias.ferraina@pix.fr>

er-lim committed:
fix: correct scalingo template

Co-authored-by: Vincent Hardouin <vincent.hardouin@pix.fr>
Co-authored-by: Matthias Ferraina <matthias.ferraina@pix.fr>
